### PR TITLE
Add geofetch module

### PR DESCRIPTION
**Pull request by mribeirodantas**
## PR checklist

Add a module for `geofetch`

- [X] This comment contains a description of changes (with reason).
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [X] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`

### CODE REVIEW

**Comment by mribeirodantas** [comment]
No idea why I'm getting `ERROR ~ Cannot enable more than one container engine -- Choose either one of: docker, singularity` in the CI/CD 😕

File: nextflow.config
Comment (Aratz) [inline nextflow.config:1]: Shouldn't this file be placed under `modules/nf-core/geofetch`? I think that's the reason you're getting the error

Comment (mribeirodantas) [reply nextflow.config:1]: Oh, thanks! I missed that 😉

**Comment by mribeirodantas** [comment]
Tests are failing due to 😞 
CleanShot 2024-03-04 at 09 45 09@2x

**Comment by mribeirodantas** [comment]
All fine now 🥳

File: modules/nf-core/geofetch/meta.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  - expression
+  - microarray
+  - sequencing
+tools:
+  - "geofetch":
+      description: "Downloads data and metadata from GEO and SRA and creates standard PEPs."
+      homepage: "http://geofetch.databio.org/"
+      documentation: "http://geofetch.databio.org/"
+      tool_dev_url: "https://github.com/pepkit/geofetch"
+      licence: ["BSD-2-clause"]
```
Comment (Aratz) [inline modules/nf-core/geofetch/meta.yml:16]:
```suggestion
      licence: ["BSD-2-clause"]
      args_id: "$args"
```

File: modules/nf-core/geofetch/meta.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  - geo_accession:
+      type: string
+      description: GEO accession ID
+
+output:
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+  - samples:
```
Comment (Aratz) [inline modules/nf-core/geofetch/meta.yml:28]: Shouldn't this be two entries? (see point 7 in https://nf-co.re/docs/contributing/modules#documentation)

File: modules/nf-core/geofetch/main.nf [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/geofetch:0.12.6--pyh7cba7a3_0':
+        'biocontainers/geofetch:0.12.6--pyh7cba7a3_0' }"
+
+    input:
+    val geo_accession
+
+    output:
+    tuple val("${geo_accession}"), path("${geo_accession}/*.CEL.gz"), emit: samples
+    path "versions.yml"           , emit: versions
```
Comment (Aratz) [inline modules/nf-core/geofetch/main.nf:14]:
```suggestion
    tuple val("${geo_accession}"), path("${geo_accession}/*.CEL.gz"), emit: samples
    path "versions.yml"                                             , emit: versions
```
Harshil Alignment™️

**Review by matthdsm — APPROVED** [review]
🚀

**Review by Aratz — COMMENTED** [review]
Looks good wrt the module guidelines. Left a couple comments for you to look at. I have never used geofetch though, so you might want to get another review from someone who's familiar with the tool.

File: modules/nf-core/geofetch/meta.yml (hunk 2 above)
Comment (mribeirodantas) [reply modules/nf-core/geofetch/meta.yml:29]: True!

**Review by Aratz — APPROVED** [review]

**Pull request merged**